### PR TITLE
PCTCompile Improvement

### PR DESCRIPTION
**Pull request by rubinhos**
Added two new parameters: ```pageSize``` and ```pageWidth``` in order to change ```PAGE-SIZE``` and ```PAGE-WIDTH``` parameter of ```COMPILE``` statement.

### CODE REVIEW

**Comment by stefandrissen** [comment]
Instead of hard-coding the progress default page-size and page-width on the java side, only write their value to the progress size if the parameter is passed.

I would also not hard-code the min / max values of the parameter on the java side.

**Comment by gquerret** [comment]
@stefandrissen +1

I've merged another feature yesterday evening, so you'll have to merge or rebase your branch. Adding my other comments directly on the files.

File: src/java/com/phenix/pct/CompilationAttributes.java [hunk 1]
```diff
@@ -415,6 +441,10 @@ protected void writeCompilationProcedure(File f, Charset c) {
             bw.newLine();
             bw.write("DEFINE INPUT PARAMETER ipListing AS CHARACTER NO-UNDO.");
             bw.newLine();
+            bw.write("DEFINE INPUT PARAMETER ipPageSize AS INTEGER NO-UNDO.");
```
Comment (gquerret) [inline src/java/com/phenix/pct/CompilationAttributes.java]: No need to pass the parameters here, as the procedure is generated on the fly, so you insert the values at line 485 only if parameters are set.

File: src/java/com/phenix/pct/CompilationAttributes.java [hunk 2]
```diff
@@ -41,6 +41,8 @@
     private boolean runList = false;
     private boolean listing = false;
     private String listingSource = null;
+    private int pageSize = 55;
```
Comment (gquerret) [inline src/java/com/phenix/pct/CompilationAttributes.java]: As @stefandrissen said, just use -1 as the default value (which would mean not set).

File: src/java/com/phenix/pct/CompilationAttributes.java [hunk 3]
```diff
@@ -452,6 +482,8 @@ else if (isUseUnderline())
             }
             if (!isXcode()) {
                 bw.write("LISTING VALUE(ipListing) ");
+                bw.write("PAGE-SIZE ipPageSize ");
```
Comment (gquerret) [inline src/java/com/phenix/pct/CompilationAttributes.java]: Include only if different from -1

File: src/progress/pct/pctBgCompile.p [hunk 4]
```diff
@@ -51,14 +51,16 @@ PROCEDURE setOptions:
...
     RUN setOption IN hComp ('LISTINGSOURCE', ENTRY(26, ipPrm, ';')).
-    RUN setOption IN hComp ('NOPARSE', IF ENTRY(27, ipPrm, ';') EQ 'true' THEN '1' ELSE '0').
-    ASSIGN lStopOnErr = ENTRY(28, ipPrm, ';') EQ 'true'.
-    RUN setOption IN hComp ('FLATTENDBG', IF ENTRY(29, ipPrm, ';') EQ 'true' THEN '1' ELSE '0').
-    RUN setOption IN hComp ('PCTDIR', ENTRY(30, ipPrm, ';')).
-    RUN setOption IN hComp ('FILELIST', ENTRY(31, ipPrm, ';')).
-    RUN setOption IN hComp ('FULLKW', IF ENTRY(32, ipPrm, ';') EQ 'true' THEN '1' ELSE '0').
-    RUN setOption IN hComp ('FULLNAMES', IF ENTRY(33, ipPrm, ';') EQ 'true' THEN '1' ELSE '0').
-    RUN setOption IN hComp ('FIELDQLF', IF ENTRY(34, ipPrm, ';') EQ 'true' THEN '1' ELSE '0').
+    RUN setOption IN hComp ('PAGESIZE', ENTRY(27, ipPrm, ';')).
```
Comment (gquerret) [inline src/progress/pct/pctBgCompile.p]: Please append the values at the end of the list instead of inserting in the middle

**Comment by gquerret** [comment]
There's probably a LF / CRLF problem in src/progress/pct/compile.p as the entire file appears to be modified.

**Comment by gquerret** [comment]
And please add a test cases for those attributes.

**Comment by rubinhos** [comment]
Hi,

I made the changes according to your feedbacks. My git config core.autocrlf was setted to false.

Tks.

**Pull request merged**